### ISSUE TITLE
Board::at() is returns PieceType::PAWN instead of PieceType::NONE on empty squares.

### ISSUE THREAD

**Issue by fqhd** [Disservin/chess-library#105]
Normally, when you run Board::at() with a Square() such as a1 or h8, you expect the Piece that is returned to have the type of the piece at that square. The problem is that this function now returns pawns on empty squares. Here is a code to test this:

Board board("8/1Q6/8/8/8/8/8/3k3K w - - 0 1");

for(int i = 0; i < 64; i++) {
	std::cout << board.at(Square(i)).type() << std::endl;
}

Considering that fen string, we expect most squares to be empty as there are only three pieces on the board, but we get mostly pawns being printed.

**Comment by Disservin** [comment]
@fqhd which version are you on? I can't replicate this.
```cpp
    Board board("8/1Q6/8/8/8/8/8/3k3K w - - 0 1");

    for (int i = 0; i < 64; i++) {
        std::cout << board.at(Square(i)).type();
    }
```
yields
```
   k   k                                         q              %                                             
```

**Comment by fqhd** [comment]
I am on version 0.6.31.

On my machine, that code yields
`pppkpppkpppppppppppppppppppppppppppppppppppppppppqpppppppppppppp%`

It might be a version issue. Let me try again with the latest version.
